c/parallel/src/jit_templates/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+file(GLOB_RECURSE template_files "*.h")
+set(jit_template "${CMAKE_CURRENT_BINARY_DIR}/jit_template.h")
+set(jit_template_src "${CMAKE_CURRENT_BINARY_DIR}/jit_template.cpp")
+set(jit_template_depfile "${CMAKE_CURRENT_BINARY_DIR}/jit_template.d")
+set(cpp_entry "${CMAKE_CURRENT_LIST_DIR}/jit_entry.h")
+
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
+
+add_custom_command(
+    OUTPUT "${jit_template}"
+    DEPENDS "${cpp_entry}" "${template_files}"
+    COMMAND "${CMAKE_CXX_COMPILER}" "${cpp_entry}" -E -o "${jit_template}"
+        -D_CCCL_C_PARALLEL_JIT_TEMPLATES_PREPROCESS
+        -MD -MT "${jit_template}" -MF "${jit_template_depfile}"
+)
+
+add_custom_command(
+    OUTPUT "${jit_template_src}"
+    DEPENDS
+      "${jit_template}"
+      "${CMAKE_CURRENT_LIST_DIR}/template_pre.h.in"
+      "${CMAKE_CURRENT_LIST_DIR}/template_post.h.in"
+    DEPFILE "${jit_template_depfile}"
```

Comment:
You want this `DEPFILE` on the first custom command since if you need it regened if any dependency changes

Reply:
D'oh, good catch, thanks!